File: java/interviewPratice/IsCryptSolution.java
public final class IsCryptSolution {

    boolean isCryptSolution(String[] crypt, char[][] solution) {
        String reg = "0+\\d+";
        for(char[] x : solution) {
            for(int i = 0; i < 3; i++) {
                if(crypt[i].contains(x[0]+"")) {
                    crypt[i] = crypt[i].replace(x[0]+"", x[1]+"");
                }
            }
        }
        if(crypt[0].matches(reg) || crypt[1].matches(reg)
            || crypt[2].matches(reg)) { return false; }
        return (Double.valueOf(crypt[0]) + Double.valueOf(crypt[1])
            == Double.valueOf(crypt[2]));
    }
}

File: javascript/interviewPractice/isCryptSolution.js
const isCryptSolution = (crypt, solution) => {
    let reg = /^0+\d+/g
    for(let x = 0; x < solution.length; x++) {
        for(let y = 0; y < 3; y++) {
            while(crypt[y].includes(solution[x][0])) {
                crypt[y] = crypt[y].replace(solution[x][0], solution[x][1])
            }
        }
    }
    if(reg.test(crypt[0]) || reg.test(crypt[1]) 
        || reg.test(crypt[2]) ) { return false }
    return (Number(crypt[0]) + Number(crypt[1]) == Number(crypt[2]))
}